app/models/unit_search_builder.rb

```diff
@@ -0,0 +1,62 @@
+# Copyright 2011-2025, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+class UnitSearchBuilder < Blacklight::SearchBuilder
+  include Blacklight::Solr::SearchBuilderBehavior
+  include Hydra::AccessControlsEnforcement
+  include Hydra::MultiplePolicyAwareAccessControlsEnforcement
+
+  self.default_processor_chain -= [:add_access_controls_to_solr_params]
+  # self.default_processor_chain += [:add_access_controls_to_solr_params_if_not_admin, :only_wanted_models, :gated_discovery_join]
+  self.default_processor_chain += [:gated_discovery_join, :only_wanted_models]
+
+  attr_accessor :user
+
+  def only_wanted_models(solr_parameters)
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] << 'has_model_ssim:"Admin::Unit"'
+  end
+
+  def add_access_controls_to_solr_params_if_not_admin(solr_parameters)
+    ability = Ability.new(user) if user.present?
+    ability ||= current_ability
+    apply_gated_discovery(solr_parameters, discovery_permissions, ability) if ability.cannot? :discover_everything, Admin::Unit
+  end
+
+  def gated_discovery_join(solr_parameters)
+    temp_solr_parameters = {}
+    add_access_controls_to_solr_params_if_not_admin(temp_solr_parameters)
+
+    query =  "{!join from=heldBy_ssim to=id}"
```

Comment:
I think we may need to doubly join here to get from the unit down to the media object.  I forget if it works to chain them:
`{!join from=heldBy_ssim to=id}{!join from=isMemberOfCollection_ssim to=id}`

Reply:
Do we need to drill down to the media object? In the current implementation, this is primarily (solely?) used by the Units index where my assumption was we would be only concerned with that child collections were being returned.